lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -196,28 +213,72 @@ def _get_device(self) -> str:
-        outputs = self.model(**inputs)
-        target_sizes = torch.tensor([image.size[::-1]])
-        results = self.processor.post_process_object_detection(outputs, target_sizes=target_sizes, threshold=threshold)[
-            0
-        ]
-        # need to wrap up the results in elements
-        elements = []
-        (w, h) = image.size
-        for score, label, box in zip(
-            results["scores"].cpu().detach().numpy(),
-            results["labels"].cpu().detach().numpy(),
-            results["boxes"].cpu().detach().numpy(),
-        ):
-            element = create_element(
-                type=self.labels[label],
-                bbox=BoundingBox(box[0] / w, box[1] / h, box[2] / w, box[3] / h).coordinates,
-                properties={"score": score},
-            )
-            elements.append(element)
-        return elements
+    def infer(
+        self, images: List[Image.Image], threshold: float, model_server_endpoint: str = ""
+    ) -> List[List[Element]]:
+        if model_server_endpoint:
+            endpoint = model_server_endpoint + self._model_name_or_path
+            metadata = {
+                "threshold": threshold,
+                "modes": [image.mode for image in images],
+                "sizes": [image.size for image in images],
+            }
+            metadata_string = json.dumps(metadata)
+            files = [("metadata", gzip.compress(metadata_string.encode("utf-8")))] + [
+                ("images", gzip.compress(image.tobytes())) for image in images
+            ]
+            try:
+                response = requests.post(endpoint, files=files)
+                results = response.json()
+            except Exception:
+                results = None
+            if results:
+                batched_results = []
+                for results, image in zip(results, images):
+                    for k, v in results.items():
+                        results[k] = base64.b64decode(v)
+                        results[k] = pickle.loads(results[k])
+                    (w, h) = image.size
+                    elements = []
+                    for score, label, box in zip(
+                        results["scores"].cpu().detach().numpy(),
+                        results["labels"].cpu().detach().numpy(),
+                        results["boxes"].cpu().detach().numpy(),
+                    ):
+                        element = create_element(
+                            type=self.labels[label],
+                            bbox=BoundingBox(box[0] / w, box[1] / h, box[2] / w, box[3] / h).coordinates,
+                            properties={"score": score},
+                        )
+                        elements.append(element)
+                    batched_results.append(elements)
+                return batched_results
+
+        batched_results = []
+
+        for image in images:
+            inputs = self.processor(images=image, return_tensors="pt").to(self._get_device())
```

Comment:
there is still those post-processing duplicate, could you merge them with above?